posts/inspecting-values-with-repl.md

```diff
@@ -0,0 +1,181 @@
+# Inspecting values with Nix REPL
+
+When working with Nix expressions, it might be useful to be able to inspect some values involved. In these cases, the Nix REPL can be pretty useful.
+
+## Some specific REPL forms
+
+When you open up a REPL, you can use `:?` to see the specific REPL forms, e.g.:
+
+```bash
+bash> nix repl
+
+Welcome to Nix version 2.2. Type :? for help.
+
+nix-repl> :?
+The following commands are available:
+
+  <expr>        Evaluate and print expression
+  <x> = <expr>  Bind expression to variable
+  :a <expr>     Add attributes from resulting set to scope
+  :b <expr>     Build derivation
+  :i <expr>     Build derivation, then install result into current profile
+  :l <path>     Load Nix expression and add it to scope
+  # ...
+```
+
+The top three entries will be most often useful for us.
+
+### Evaluate
+
+The first is the same as an expression evaluation:
+
+```bash
+nix-repl> 1
+1
+```
+
+### Bind
+
+The second is the same as a `let`-bind in an expression:
+
+```bash
+nix-repl> a = 1
+
+nix-repl> a
+1
+```
+
+This is equivalent to `let a = 1; in ...`
```

Comment:
This might make me think that I can only use it in the next line, particularly when combined with the next section, but what this actually does is add it to scope right? I guess you could argue that `...` is the rest of the session, but not sure that is completely clear.

Reply:
I guess the fact that the next section explicitly names `attributes` makes it reasonably clear about the distinction, but still a bit uncertain.

Reply:
yeah, might as well put that in